### PR TITLE
Fix init of block settings

### PR DESCRIPTION
**Pull request by core23**
## Subject

This is a bug fix for the usage without the page bundle.

```twig
    {{ sonata_block_render({ 'type': 'sonata.media.block.gallery' }, {
        // ...
        "galleryId": 2
    }) }}
```

When passing a block definition `$meta` array to this method, the `$settings` will be ignored for this block itself, they are only passed to the block context. This will result in a null value for different blocks (e.g. https://github.com/sonata-project/SonataMediaBundle/blob/9946d8ac3e0f21f1198050e5f7501766b058e53a/src/Block/GalleryBlockService.php#L169).

I am targeting this branch, because this is a patch.

## Changelog

```markdown
### Fixed
- Fix init of block settings
```

### CODE REVIEW

File: src/Block/BlockContextManager.php [hunk 1]
```diff
@@ -134,6 +134,8 @@ public function get($meta, array $settings = []): BlockContextInterface
                 // merge user settings
                 $settings = array_merge($meta['settings'], $settings);
             }
+
+            $block->setSettings($settings);
```
Comment (VincentLanglet) [inline src/Block/BlockContextManager.php:138]: Are we supposed to also set the Settings in the `else` part, this would be consistent no ?

Comment (core23) [reply src/Block/BlockContextManager.php:138]: Not sure about this. When you pass a `BlockInterface` in the else part your block could already have valid settings.

Comment (VincentLanglet) [reply src/Block/BlockContextManager.php:138]: We doing later
```
try {
            $settings = $this->resolve($block, array_merge($block->getSettings(), $settings));
        } catch (ExceptionInterface $e) {
```
So this means that the $settings might not be compatible with the Block provided.

That's why I wonder if we shouldn't set the settings after the try/catch call

**Comment by VincentLanglet** [comment]
@jordisala1991 You worked a little with blocks when upgrading SonataPageBundle, you might know this better than me

**Comment by VincentLanglet** [comment]
Friednly ping @jordisala1991

**Review by jordisala1991 — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**